models/reviews.js

```diff
@@ -1,19 +1,29 @@
 const db = require("../db/connection");
 
 exports.selectReview = (req) =>{
-    return db.query(`SELECT review_id, title, review_body, designer, review_img_url, votes, category, owner, created_at
+    return db.query(`
+    SELECT reviews.*, COUNT(reviews.review_id)::INT AS comment_count
     FROM reviews
-    WHERE review_id = $1`, [req.params.review_id]).then((review) =>{
-        if(review.rows.length === 0){
+    JOIN comments ON reviews.review_id = comments.review_id
+    WHERE reviews.review_id = $1
+    GROUP BY reviews.review_id`, [req.params.review_id]).then((review) =>{
+        return Promise.all([selectJustReview(req), review])
+    }).then(([reviewForId, reviewWithComments]) =>{
+        //if there isn't a review for that id
```

Comment:
Here you are invoking your `selectJustReview` function and then if there is no single review returned rejecting the promise. 
However, are you not already handling this behaviour in the behaviour already written for `selectJustReview`? 
If there is no review, an error is being thrown, so if you are entering the .then block on line 11, there _must_ be a review matching that id.